client/gutenberg/editor/components/post-publish-button-or-toggle/index.jsx

```diff
@@ -0,0 +1,225 @@
+		) {
+			this.togglePostSaving();
+		}
+	}
+
+	togglePostSaving() {
+		const {
+			userNeedsVerification,
+			lockPostSaving,
+			unlockPostSaving,
+			createWarningNotice,
+			setState,
+			verificationNoticeLabel,
+		} = this.props;
+
+		const lockName = 'blockEditorPostSavingLock';
+
+		if ( userNeedsVerification ) {
+			lockPostSaving( lockName );
+
+			createWarningNotice( verificationNoticeLabel, {
+				id: 'verify-email-notice',
+				isDismissible: false,
+				actions: [
+					{
+						label: translate( 'Learn More' ),
+						url: '#',
+						onClick: e => {
+							e.preventDefault();
+							setState( { showEmailVerificationNotice: true } );
+						},
+					},
+				],
+			} );
+		} else {
+			unlockPostSaving( lockName );
+		}
+	}
+
+	closeVerifyEmailDialog = () => {
+		this.props.setState( { showEmailVerificationNotice: false } );
+	};
+
+	render() {
+		const {
+			forceIsDirty,
+			forceIsSaving,
+			hasPublishAction,
+			isBeingScheduled,
+			isLessThanMediumViewport,
+			isPending,
+			isPublished,
+			isPublishSidebarEnabled,
+			isPublishSidebarOpened,
+			isScheduled,
+			togglePublishSidebar,
+			showEmailVerificationNotice,
+		} = this.props;
+
+		const IS_TOGGLE = 'toggle';
+		const IS_BUTTON = 'button';
+		let component;
+
+		/**
+		 * Conditions to show a BUTTON (publish directly) or a TOGGLE (open publish sidebar):
+		 *
+		 * 1) We want to show a BUTTON when the post status is at the _final stage_
+		 * for a particular role (see https://codex.wordpress.org/Post_Status):
+		 *
+		 * - is published
+		 * - is scheduled to be published
+		 * - is pending and can't be published (but only for viewports >= medium).
+		 * 	 Originally, we considered showing a button for pending posts that couldn't be published
+		 * 	 (for example, for an author with the contributor role). Some languages can have
+		 * 	 long translations for "Submit for review", so given the lack of UI real estate available
+		 * 	 we decided to take into account the viewport in that case.
+		 *  	 See: https://github.com/WordPress/gutenberg/issues/10475
+		 *
+		 * 2) Then, in small viewports, we'll show a TOGGLE.
+		 *
+		 * 3) Finally, we'll use the publish sidebar status to decide:
+		 *
+		 * - if it is enabled, we show a TOGGLE
+		 * - if it is disabled, we show a BUTTON
+		 */
+		if (
+			isPublished ||
+			( isScheduled && isBeingScheduled ) ||
+			( isPending && ! hasPublishAction && ! isLessThanMediumViewport )
+		) {
+			component = IS_BUTTON;
+		} else if ( isLessThanMediumViewport ) {
+			component = IS_TOGGLE;
+		} else if ( isPublishSidebarEnabled ) {
+			component = IS_TOGGLE;
+		} else {
+			component = IS_BUTTON;
+		}
+
+		return (
+			<Fragment>
+				<PostPublishButton
+					forceIsDirty={ forceIsDirty }
+					forceIsSaving={ forceIsSaving }
+					isOpen={ isPublishSidebarOpened }
+					isToggle={ component === IS_TOGGLE }
+					onToggle={ togglePublishSidebar }
+				/>
+				{ showEmailVerificationNotice && (
```

Comment:
Since we have an actual email verification notice in this component, I'd definitely change this into `showEmailVerificationDialog` to avoid confusion.